lib/cloud-id-shim.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+RUN_CLOUD_INIT=/run/cloud-init
+if [ ! -d "$RUN_CLOUD_INIT" ]; then
+  exit 0
+fi
+
+CLOUD_ID_LINK_PATH=$RUN_CLOUD_INIT/cloud-id
+if [ -L "$CLOUD_ID_LINK_PATH" ]; then
+  exit 0
+fi
+
+CLOUD_ID=$(cloud-id 2>/dev/null) || CLOUD_ID=""
+if [ -z "$CLOUD_ID" ]; then
+  exit 0
+fi
+
+CLOUD_ID_FILE_PATH=$RUN_CLOUD_INIT/cloud-id-$CLOUD_ID
+
+echo "$CLOUD_ID" > "$CLOUD_ID_FILE_PATH"
+ln -s "$CLOUD_ID_FILE_PATH" "$CLOUD_ID_LINK_PATH"
```

Comment:
Is it worth a message "Creating symlink X->Y" so we have record in `systemctl status ubuntu-advantage-cloud-id-shim.service` that the symlink was actually created? It only makes it easier to confirm that shim actually did something up in systemctl (even though we could just go and ls /run/cloud-init/cloud-id*)